instrumentation/opentelemetry_broadway/lib/opentelemetry_broadway.ex

```diff
@@ -137,6 +153,58 @@ defmodule OpentelemetryBroadway do
     OpentelemetryTelemetry.end_telemetry_span(@tracer_id, metadata)
   end
 
+  @doc false
+  def handle_batch_start(
+    _event,
+    _measurements,
+    %{
+      topology_name: topology_name,
+      messages: messages,
+      batch_info: %Broadway.BatchInfo{batcher: batcher},
+    } = metadata,
+    _config
+  ) do
+    span_name = "#{inspect(topology_name)}/#{Atom.to_string(batcher)} batching process"
+
+    attributes = %{
+      :"broadway.batch.count" => length(messages)
+    }
+
+    OpentelemetryTelemetry.start_telemetry_span(@tracer_id, span_name, metadata, %{
+      kind: :internal,
+      attributes: attributes
+    })
+  end
+
+  @doc false
+  def handle_batch_stop(
+    _event,
+    _measurements,
+    %{
+      successful_messages: successful_messages,
+      failed_messages: failed_messages
+    } = metadata,
+    _config
+  ) do
+    status = if length(failed_messages) == 0 do
+      OpenTelemetry.status(:ok)
+    else
+      OpenTelemetry.status(:error, "Batch completed with failed messages")
+    end
```

Comment:
`length/1` always calculates the whole length. Do this instead:

```suggestion
    status = if Enum.empty?(failed_messages) do
      OpenTelemetry.status(:ok)
    else
      OpenTelemetry.status(:error, "Batch completed with failed messages")
    end
```

which will short-circuit instead.